webp_backport/src/main/java/de/marcreichelt/webp_backport/FileSignatureChecker.java

```diff
@@ -0,0 +1,59 @@
+package de.marcreichelt.webp_backport;
+
+class FileSignatureChecker {
+
+    /**
+     * Check if a given byte array has the necessary WebP signature in the beginning of its content
+     *
+     * @param imgByteArr the byte array to check
+     * @return true if the byte array has the WebP signature else false
+     */
+    static boolean checkForWebP(byte[] imgByteArr) {
+        if (byteArrayStartsWith(imgByteArr, new byte[]{82, 73, 70, 70})) { // bytes 1 to 3 must be 'RIFF'
+
+            // bytes 4 to 7 == file length information, this info differs every time so don't check
+
+            // bytes 8 to 11 must be 'WEBP'
+            if (imgByteArr.length >= 12) {
+                byte[] webp = new byte[]{87, 69, 66, 80}; // 'WEBP'
+                for (int i = 8; i <= 11; i++) {
+                    if (imgByteArr[i] != webp[i - 8]) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
This method is useful only for WebP images, but has nearly everything it needs to do a general check for a RIFF format according to https://en.wikipedia.org/wiki/List_of_file_signatures
Maybe we could create another method

    boolean isRIFF(byte[] data, byte[]fourCC)

which we then use with the 'WEBP' bytes to detect if it's a WebP image.